### PR TITLE
Add missing 2021 Idaho 5.5% income tax bracket

### LINKED ISSUE #3127: Add missing 2021 Idaho 5.5% income tax bracket
It appears that the joint and widow parameter files are missing the 2021 5.5% tax bracket even though the cited reference in the Idaho parameter files shows the 5.5% tax bracket.

### PR DESCRIPTION
**Pull request by WeihaoZ724**
Fixes #3127

### Summary
🆕💵💑

This pull request updates the income tax parameters for Idaho to reflect the latest tax rates for 2021. It adds a new tax bracket for `joint` and `widow` filers with a 5.5% rate in the files `joint.yaml` and `widow.yaml`.

> _`joint` and `widow(er)`_
> _new tax bracket in Idaho_
> _spring brings higher rates_

### Walkthrough
* Add a new income tax bracket for joint and widow(er) filers in Idaho ([link](https://github.com/PolicyEngine/policyengine-us/pull/3142/files?diff=unified&w=0#diff-55eedf5ab87c7c12dfb3548a65afc5fbd27b4c52735fdb22f864ccf3240ca7a2L36-R38), [link](https://github.com/PolicyEngine/policyengine-us/pull/3142/files?diff=unified&w=0#diff-10a742b8b1887809cc0436ba04cf02bc29e7c07ba1cf5f8a7dc09961ef540a05R36-R39))

### CODE REVIEW

File: policyengine_us/parameters/gov/states/id/tax/income/main/joint.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/id/tax/income/main/joint.yaml:1]: add changelog_entry

File: policyengine_us/parameters/gov/states/id/tax/income/main/joint.yaml [hunk 1]
```diff
@@ -33,3 +33,7 @@ brackets:
     rate:
       2021-01-01: 0.065
       2022-01-01: 0.06
+  - threshold:
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/id/tax/income/main/joint.yaml:36]: add bracket in the middle of file

File: policyengine_us/parameters/gov/states/id/tax/income/main/widow.yaml [hunk 2]
```diff
@@ -33,3 +33,7 @@ brackets:
     rate:
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/id/tax/income/main/widow.yaml:33]: add bracket in the middle of file

File: policyengine_us/parameters/gov/states/id/tax/income/main/joint.yaml (hunk 1 above)
Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/id/tax/income/main/joint.yaml:36]:
```
    rate:
      2021-01-01: 0.045
  - threshold:
      2021-01-01: 12_702
      2022-01-01: 16_622
    rate:
      2021-01-01: 0.055
      2022-01-01: 0.06
  - threshold:
      2021-01-01: 15_878
      2022-01-01: .inf
    rate:
      2021-01-01: 0.065

```

File: policyengine_us/parameters/gov/states/id/tax/income/main/joint.yaml
Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/id/tax/income/main/joint.yaml:1]: add a tests case to `id_income_tax_before_refundable_credits`

File: changelog_entry.yaml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+- bump: minor
```
Comment (PavelMakarchuk) [inline changelog_entry.yaml:1]:
```suggestion
- bump: patch
```

File: changelog_entry.yaml [hunk 4]
```diff
@@ -0,0 +1,4 @@
...
+  changes:
+    added:
```
Comment (PavelMakarchuk) [inline changelog_entry.yaml:3]:
```suggestion
    fixed:
```

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**